lib/groups.js

```diff
@@ -191,38 +332,148 @@ module.exports.createGroup = function (groupName, assessmentId, userId, authnUse
   }
 };
 
-module.exports.leaveGroup = function (assessmentId, userId, authnUserId, callback) {
-  const params = [assessmentId, userId, authnUserId];
-  sqldb.call('group_leave', params, function (err) {
-    if (ERR(err, callback)) return;
-    callback();
+/**
+ * @param {string} assessmentId
+ * @param {string} userId
+ * @param {string} authnUserId
+ */
+module.exports.leaveGroup = async function (assessmentId, userId, authnUserId) {
+  await sqldb.runInTransactionAsync(async () => {
+    const groupId = await module.exports.getGroupId(assessmentId, userId);
+    const groupConfig = await module.exports.getGroupConfig(assessmentId);
+
+    if (groupConfig.has_roles) {
+      const groupInfo = await module.exports.getGroupInfo(groupId, groupConfig);
+
+      // Reassign roles if there is more than 1 user
```

Comment:
Now that we're doing just a little bit of DB I/O at the start and end of this function, all this role reassignment feels like it could be pretty easily unit-tested if it were split into a separate function!

Reply:
Would we need to unit-test this if we've already testing the functionality of this logic in `groupRole.test.js` (which is more of an integration test I think)? That said, I can see how this is enough code to warrant being in its own function.